security/oc_pstat.c

```diff
@@ -249,24 +265,32 @@ pstat_check_state(const oc_sec_pstat_t *ps, size_t device)
 
 static bool
 oc_pstat_handle_state(oc_sec_pstat_t *ps, size_t device, bool from_storage,
-                      bool close_all_tls_connections_immediately)
+                      bool shutdown)
 {
   OC_DBG("oc_pstat: Entering pstat_handle_state");
   switch (ps->s) {
   case OC_DOS_RESET: {
+    // reset is in progress
+    if (g_pstat[device].reset_in_progress) {
+      OC_DBG("oc_pstat: reset in progress");
+      return false;
+    }
+    g_pstat[device].reset_in_progress = true;
+    oc_remove_delayed_callback((void *)device, delayed_reset);
     ps->p = true;
     ps->isop = false;
     ps->cm = 1;
     ps->tm = 2;
     ps->om = 3;
     ps->sm = 4;
+
+    memset(ps->rowneruuid.id, 0, sizeof(ps->rowneruuid.id));
 #if defined(OC_SERVER) && defined(OC_CLIENT) && defined(OC_CLOUD)
-    cloud_reset(device, true, 0);
+    cloud_reset(device, true, false, 0);
```

Comment:
so the sync parameter was changed from `true` to `false`, meaning that cloud_manager will attempt to login (if the device was not logged in) can that ever succeed since the request will be generated and process after this function finishes?

If it is correct then the TODO note can be removed since, `async` cloud_reset is now allowed.

Reply:
there is new parameter in `cloud_reset(device, force, async, ...)` if force is set the deregistration is omitted (`async` flag is ignored)